warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1188,22 +1194,29 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
             
             // Expand any bounded ranges into a conjunction of discrete terms
             try {
-                innerStopwatch = timers.newStartedStopwatch("DefaultQueryPlanner - Expand regex");
                 ParallelIndexExpansion regexExpansion = new ParallelIndexExpansion(config, scannerFactory, metadataHelper, expansionFields,
                                 config.isExpandFields(), config.isExpandValues(), config.isExpandUnfieldedNegations());
-                queryTree = (ASTJexlScript) regexExpansion.visit(queryTree, null);
-                if (log.isDebugEnabled()) {
-                    logQuery(queryTree, "Query after expanding regex:");
+                
+                // Check if there is any regex to expand.
+                if (NodeTypeCountVisitor.countNodes(queryTree).hasAny(ASTNRNode.class, ASTERNode.class)) {
```

Comment:
Do we have to call the visitor every time or can we count the nodes once?  There may be places where we need to recount but certainly not at line 1211

Reply:
@ivakegg I made a modification so that the nodes are recounted only when we know the query tree has been modified. Let me know if this works for you.

Reply:
I believe we are still calling it too often.  If we expand regexs and ranges, that will only produce more equality nodes and maybe ExceededValue markers.  I don't think any of the subsequent calls actually depend on those counts so why recount?

Reply:
Edited the code to only count the nodes twice, once before the first round of expanding regex and ranges, and once before the second round of expanding regex and ranges after pulling up delayed predicates.